Automattic/simplenote-ios
# Passkeys: UI interaction feedback and errors

### Fix
Soooo... the first PR for implementing passkeys into iOS has some user interaction issues.  None of the errors are handled, so if it fails it just does nothing.... and the registration and auth processes happen over time and can be slow, but the UI give no activity indication, and isn't locked in anyway so you can keep poking and clicking all over the place... not great....

So I have fixed that in this PR.  Here you will see:

- When registering a passkey a full screen modal spinner appears
- if registration succeeds you see a success message
- if registration fails you see an error
- When logging in using passkeys the ui locks and the sign in button activity indicator animates
- if auth fails you will see an alert

### Test
Registration:
1. Log into an account with email and password
2. go to Settings and tap on Add Passkey authentication. Confirm that you see a spinner
3. When the passkey modal pops up, dismiss it.  Confirm you see an error message
4. Tap on the app passkey button again and this time complete the registration.  Confirm you see a success message (note if you already have a passkey registered you will need to delete it from the [server](https://console.cloud.google.com/datastore/databases/-default-/entities;kind=WebauthnCredential;ns=__$DEFAULT$__/query/kind?project=simple-note-hrd)

Auth:
1. log out of your account
2. tap on login -> Login with passkeys
3. enter an email address that isn't valid.  Confirm no login attempt is done and you get an alert that the email is invalid
4. enter the email address you registered with.  When the passkey modal pops up, dismiss it.  You should see an alert and the ui should unlock
5. Try logging in again with the correct email and this time confirm the passkey auth.  Confirm you get logged in and the ui unlocks


### Review
***(Required)*** Add instructions for reviewers.  For example:
> Only one developer and one designer are required to review these changes, but anyone can perform the review.

### Release
***(Required)*** Add a concise statement to `RELEASE-NOTES.txt` if the changes should be included in release notes. Include details about updating the notes in this section. For example:
> `RELEASE-NOTES.txt` was updated in d3adb3ef with:
> 
> > Added markdown support

If the changes should not be included in release notes, add a statement to this section. For example:

> These changes do not require release notes.
